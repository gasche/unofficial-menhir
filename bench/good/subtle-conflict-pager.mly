%token A B C D L R
%type <unit> s
%start s

/* Cf. mon journal au 19 septembre 2005. Si on résoud le conflit situé
   dans l'état LA en donnant priorité à la réduction, alors la chaîne
   RABC, qui aurait dû être acceptée sans ambiguïté, est rejetée. */

%nonassoc B
%nonassoc A /* gives higher precedence to production b -> A over token B */

%%

s:
  L l { () }
| R r { () }

l:
  a C { () }
| b B { () }

r:
  a C { () }
| b D { () }

a:
  A B { () }

b:
  A   { () }

