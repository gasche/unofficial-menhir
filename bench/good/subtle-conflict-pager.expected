%start s
%token R
%token L
%token D
%token C
%token B
%token A
%nonassoc B 
%nonassoc A 
%type <unit> s
%%

s:
| L l
    { () }
| R r
    { () }

l:
| a C
    { () }
| b B
    { () }

r:
| a C
    { () }
| b D
    { () }

a:
| A B
    { () }

b:
| A
    { () }

%%



